src/PresentationalComponents/Modals/AddEditTopic.js

```diff
@@ -0,0 +1,193 @@
+import './_AddEditTopic.scss';
+
+import React, { useState, useEffect, useRef } from 'react';
+import { injectIntl } from 'react-intl';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+    Modal, Form, FormGroup, Button, Split,
+    SplitItem, TextInput, TextArea, Radio, Checkbox
+} from '@patternfly/react-core';
+
+import messages from '../../Messages';
+import { setTopic, deleteTopic } from '../../AppActions';
+
+const AddEditTopic = ({ handleModalToggle, intl, isModalOpen, topic, setTopic, deleteTopic }) => {
+
+    const [name, setName] = useState();
+    const [description, setDescription] = useState();
+    const [tag, setTag] = useState();
+    const [enabled, setEnabled] = useState(false);
+    const [featured, setFeatured] = useState();
+
+    const setDeleteHandler = ({ type, options }) => {
+        type === 'DELETE' ?
+            deleteTopic(options) :
+            setTopic(options);
+        handleModalToggle(false);
+    };
+
+    const footer = (
+        <Split gutter='sm' style={{ width: '100%' }}>
+            <SplitItem>
+                <Button key='confirm' variant='primary' onClick={ () => setDeleteHandler({
+                    type: 'SET', options: { slug: tag, name, description, enabled, featured }
+                })}>
+                    {intl.formatMessage(messages.save)}
+                </Button>
+            </SplitItem>
+            <SplitItem>
+                <Button key='cancel' variant='secondary' onClick={ () => handleModalToggle(false) }>
+                    {intl.formatMessage(messages.cancel)}
+                </Button>
+            </SplitItem>
+            <SplitItem isFilled/>
+            <SplitItem>
+                <Button key='delete' variant='danger' onClick={ () => setDeleteHandler({
+                    type: 'DELETE', options: { slug: tag, name, description, enabled, featured }
+                })}>
+                    {intl.formatMessage(messages.deleteTopic)}
+                </Button>
+            </SplitItem>
+        </Split>
+    );
+
+    const insertTopicValues = () => {
```

Comment:
from here to 76, can yah walk me through what we're doing here?  (i :suspect:  we can greatly simplify da task ) (maybe over 👖 )

Reply:
So I can remove the `insertTopicValues` and move this to the `useState` for each value and then use the setter methods for each in the effect on line 72. For the other two methods, I was using these to compare the previous state to the current state so I know when to update the values of the topic, which then updates the values that are actually displayed. I wasn't 100% sure how to make that work, so you probably will have a better solution 😆

Reply:
dayum. late worker, they must be payin yah over time 😏  we'll do ah 👖 tomorrow! whenever yer around n talk about it 💃